latch/registry/table.py

```diff
@@ -0,0 +1,257 @@
+
+    @classmethod
+    def from_name(
+        cls,
+        project: Project,
+        table_name: str,
+    ):
+        data = gql.execute(
+            document="""
+                query tableInfoQuery ($argProjectId: BigInt!) {
+                    catalogExperiments(condition: {
+                        projectId: $argProjectId,
+                        removed: false
+                    }) {
+                        nodes {
+                            id,
+                            displayName
+                        }
+                    }
+                }
+            """,
+            variables={"argProjectId": project.id},
+        )["catalogExperiments"]["nodes"]
+
+        for experiment in data:
+            if experiment["displayName"] == table_name:
+                return cls(table_id=experiment["id"])
+
+        raise ValueError(f"unable to find table {table_name} in project {project.name}")
+
+    @property
+    def name(self):
+        if self._name is not None:
+            return self._name
+
+        self._name = gql.execute(
+            """
+            query tableNameQuery ($argTableId: BigInt!) {
+                catalogExperiment(id: $argTableId) {
+                    displayName
+                }
+            }
+            """,
+            variables={
+                "argTableId": self.id,
+            },
+        )["catalogExperiment"]["displayName"]
+
+        return self._name
+
+    @property
+    def columns(self):
+        if self._columns is not None:
+            return self._columns
+
+        self._columns = gql.execute(
+            """
+            query tableColumnsQuery ($argTableId: BigInt!) {
+                catalogExperiment(id: $argTableId) {
+                    catalogExperimentColumnDefinitionsByExperimentId {
+                        nodes {
+                            key
+                            type
+                        }
+                    }
+                }
+            }
+            """,
+            variables={"argTableId": self.id},
+        )["catalogExperiment"]["catalogExperimentColumnDefinitionsByExperimentId"][
+            "nodes"
+        ]
+
+        return self._columns
+
+    @property
+    def row_type(self):
+        if self._row_type is not None:
+            return self._row_type
+
+        fields = []
+        for column in self.columns:
+            cleaned = clean(column["key"])
+            column_db_type = column["type"]
+
+            fields.append(
+                (
+                    cleaned,
+                    to_python_type(
+                        column_db_type["type"],
+                        cleaned,
+                        column_db_type["allowEmpty"],
+                    ),
+                )
+            )
+
+        return make_dataclass(
+            clean(f"{self.name}_type"),
+            fields,
+            bases=(Row,),
+            frozen=True,
+        )
+
+    def list(self) -> List[Row]:
```

Comment:
should be a generator that yields pages of results

APIs should not return lists of things that can be very large (1000s of elements and/or >1MB total bytes)

Reply:
should accept a keyword-only filter for names (I'd do `equals`, `like`, `ilike` and `in` + the `not` variants of those)